pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1157,7 +1157,11 @@ protected void handleSubscribe(final CommandSubscribe subscribe) {
                                         remoteAddress, topicName, subscriptionName);
                                 commandSender.sendSuccessResponse(requestId);
                                 if (brokerInterceptor != null) {
-                                    brokerInterceptor.consumerCreated(this, consumer, metadata);
+                                    try {
+                                        brokerInterceptor.consumerCreated(this, consumer, metadata);
```

Comment:
I am not sure if we need to send success before catching this exception. if it throws RuntimeException, maybe we should return an error first, not `sendSuccessResponse`.

Reply:
The exception thrown by an interceptor does not mean an error from broker, it's an error from 3rd party extensions. A similar case is that you can see `ConsumerInterceptors` , which catches all `Throwable`s from the user-configured interceptor.

Reply:
The key problem now is the BrokerInterceptor exception is caught by the broker and then sends a failed response with a duplicated requestId.  So we can't get any information about the exception. 

For the exception processing, I agree with @BewareMyPower. It's better to catch`Interceptors` exceptions than throw them and return an error.